Sources/SwiftKueryPostgreSQL/PostgreSQLConnection.swift

```diff
@@ -418,84 +464,95 @@ public class PostgreSQLConnection: Connection {
             // Since we set the single row mode, PGRES_TUPLES_OK means the result is empty, i.e. there are
             // no rows to return.
             clearResult(result, connection: self)
-            onCompletion(.successNoData)
+            runCompletionHandler(.successNoData, onCompletion: onCompletion)
         }
         else if status == PGRES_SINGLE_TUPLE {
             let resultFetcher = PostgreSQLResultFetcher(queryResult: result, connection: self)
             setState(.fetchingResultSet)
             currentResultFetcher = resultFetcher
-            onCompletion(.resultSet(ResultSet(resultFetcher)))
+            runCompletionHandler(.resultSet(ResultSet(resultFetcher, connection: self)), onCompletion: onCompletion)
         }
         else {
             let errorMessage = String(validatingUTF8: PQresultErrorMessage(result)) ?? "Unknown"
             clearResult(result, connection: self)
-            onCompletion(.error(QueryError.databaseError("Query execution error:\n" + errorMessage + " For query: " + query)))
+            runCompletionHandler(.error(QueryError.databaseError("Query execution error:\n" + errorMessage + " For query: " + query)), onCompletion: onCompletion)
         }
     }
-    
-    
+
     /// Start a transaction.
     ///
     /// - Parameter onCompletion: The function to be called when the execution of start transaction command has completed.
     public func startTransaction(onCompletion: @escaping ((QueryResult) -> ())) {
-        executeTransaction(command: "BEGIN", inTransaction: false, changeTransactionState: true, errorMessage: "Failed to rollback the transaction", onCompletion: onCompletion)
+        DispatchQueue.global().async {
```

Comment:
Similar comment to execute, let's move the offload into executeTransaction so there are fewer places to change in the future.